### PR TITLE
fix reloc support for ppc, and ignore diffs to symbols with catch all symbols

### PR DESCRIPTION
**Pull request by snuffysasa**
Making this a draft pull request, because I still have questions and I assume you will have feedback.

### CODE REVIEW

File: src/objdump.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ class ArchSettings:
     re_reg: Pattern[str]
     re_sprel: Pattern[str]
     re_includes_sp: Pattern[str]
+    re_reloc: str
```
Comment (simonlindholm) [inline src/objdump.py:35]: the name `re_reloc` indicates that it's a Pattern[str]

File: src/objdump.py [hunk 2]
```diff
@@ -150,6 +155,81 @@ def parse_relocated_line(line: str) -> Tuple[str, str, str]:
     return before, imm, after
 
 
+def process_reloc(reloc_row: str, prev: str):
+    if prev == "<skipped>":
+        return -1
```
Comment (simonlindholm) [inline src/objdump.py:160]: -1 is a bad sentinel, None please

File: src/objdump.py [hunk 3]
```diff
@@ -150,6 +155,81 @@ def parse_relocated_line(line: str) -> Tuple[str, str, str]:
     return before, imm, after
 
 
+def process_reloc(reloc_row: str, prev: str):
```
Comment (simonlindholm) [inline src/objdump.py:158]: return type; I think `mypy` will complain as is, please check

File: src/scorer.py [hunk 4]
```diff
@@ -55,6 +53,18 @@ def score(self, cand_o: Optional[str]) -> Tuple[int, str]:
...
 
+        def imm_matches_everything(row: str, arch: ArchSettings) -> bool:
+            if arch.name == "ppc":
+                return any(str in row for str in ["...data", "@"])
+            else:
+                return "(." in row
+
+        def is_valid_symbol(str: str) -> bool:
+            return (
+                re.sub(re.compile(r"[A-Za-z_$][A-Za-z0-9_$]*"), "", str.split("@")[0])
```
Comment (simonlindholm) [inline src/scorer.py:64]: there are functions in `re` for checking whether a regex matches another string, you don't have to do search-replace

File: src/objdump.py (hunk 1 above)
Comment (snuffysasa) [reply src/objdump.py:35]: good point, I'll fix

File: src/objdump.py (hunk 2 above)
Comment (snuffysasa) [reply src/objdump.py:160]: k

File: src/objdump.py (hunk 3 above)
Comment (snuffysasa) [reply src/objdump.py:158]: ohh, I thought black was auto adding return types... but yea I need to install, mypy. I'll do that and add return type

File: src/scorer.py (hunk 4 above)
Comment (snuffysasa) [reply src/scorer.py:64]: on it

**Comment by snuffysasa** [comment]
I made a replacement PR on a new branch that I made, closing this one

**Pull request closed without merging**